
On OmniOS, several libraries have had their functionality moved into libc
but the old libraries persist as stubs for compatibility reasons. Two such
libraries are libdl.so & libpthread.so

There's no specific problem with an application or library linking with these
(just a slight overhead) but when a library is used by a component within
illumos-omnios, a check is made which generates a warning if an
unecessary library is linked.

Openssl is one of the libraries used by illumos-omnios and therefore we
patch the configuration script to stop linking against these two libraries.

diff -wpruN '--exclude=*.orig' a~/Configurations/10-main.conf a/Configurations/10-main.conf
--- a~/Configurations/10-main.conf	1970-01-01 00:00:00
+++ a/Configurations/10-main.conf	1970-01-01 00:00:00
@@ -179,7 +179,7 @@ sub vms_info {
         inherit_from     => [ "BASE_unix" ],
         template         => 1,
         cflags           => "-DFILIO_H",
-        ex_libs          => add("-lsocket -lnsl -ldl"),
+        ex_libs          => add("-lsocket -lnsl"),
         dso_scheme       => "dlfcn",
         thread_scheme    => "pthreads",
         shared_target    => "solaris-shared",
@@ -202,9 +202,7 @@ sub vms_info {
         cc               => "gcc",
         cflags           => add_before(picker(default => "-Wall -DL_ENDIAN -DOPENSSL_NO_INLINE_ASM",
                                               debug   => "-O0 -g",
-                                              release => "-O3 -fomit-frame-pointer"),
-                                       threads("-pthread")),
-        ex_libs          => add(threads("-pthread")),
+                                              release => "-O3 -fomit-frame-pointer")),
         bn_ops           => "BN_LLONG",
         shared_cflag     => "-fPIC",
         shared_ldflag    => "-shared -static-libgcc",
@@ -222,9 +220,7 @@ sub vms_info {
         cc               => "gcc",
         cflags           => add_before(picker(default => "-m64 -Wall -DL_ENDIAN",
                                               debug   => "-O0 -g",
-                                              release => "-O3"),
-                                       threads("-pthread")),
-        ex_libs          => add(threads("-pthread")),
+                                              release => "-O3")),
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         perlasm_scheme   => "elf",
         shared_cflag     => "-fPIC",
